### PR TITLE
[react-native] Add missing GlobalFetch / RequestInfo definitions

### PR DESCRIPTION
**Pull request by VincentLanglet**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://facebook.github.io/react-native/docs/network.html
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

In the continuation of https://github.com/DefinitelyTyped/DefinitelyTyped/pull/25255

### CODE REVIEW

**Review by alloy — CHANGES_REQUESTED** [review]
This looks good to me, but could you please add some test coverage for these changes?

**Comment by VincentLanglet** [comment]
@alloy Is this good ?

File: types/react-native/test/globals.tsx [hunk 1]
```diff
@@ -0,0 +1,3 @@
+const fetchCopy: GlobalFetch['fetch'] = fetch;
+
+const requestInfoAsString: RequestInfo = 'request';
```
Comment (alloy) [inline types/react-native/test/globals.tsx]: Please use real world code examples, e.g. taken from [here](https://github.com/mdn/fetch-examples) or possibly React Native examples, and preferably without explicit typing where possible.

**Comment by VincentLanglet** [comment]
@alloy Thanks for your advices.
I tried to do this PR to fix an error I have in the apollo-link-http-common modules but i do not usually use the fetch function so I didn't know how to test this. Is it better now ? I tried to use your examples.

**Review by alloy — APPROVED** [review]

**Comment by alloy** [comment]
Ace, thanks! 👍

**Pull request merged**